MeesGame/Gameplay/GameObjects/Player.cs

```diff
@@ -210,26 +239,38 @@ public HumanPlayer(Level level, TileField tileField, Point location, int layer =
 
         public override void HandleInput(InputHelper inputHelper)
         {
-            if (inputHelper.IsKeyDown(Keys.W) || inputHelper.IsKeyDown(Keys.Up))
-            {
-                nextAction = NORTH;
-            }
-            else if (inputHelper.IsKeyDown(Keys.D) || inputHelper.IsKeyDown(Keys.Right))
-            {
-                nextAction = EAST;
-            }
-            else if (inputHelper.IsKeyDown(Keys.S) || inputHelper.IsKeyDown(Keys.Down))
-            {
-                nextAction = SOUTH;
-            }
-            else if (inputHelper.IsKeyDown(Keys.A) || inputHelper.IsKeyDown(Keys.Left))
+            if (this.currentlyMoving == false)
             {
-                nextAction = WEST;
-            }
-            else
-            {
-                nextAction = NONE;
+                if (inputHelper.IsKeyDown(Keys.W) || inputHelper.IsKeyDown(Keys.Up))
+                {
+                    nextAction = NORTH;
+                    //this.sprite = new SpriteSheet("playerup@4", 0);
+                    walkAnimation("playerup@4");
```

Comment:
You want to start the animation when the timedplayer starts moving, not when handling input. Makes it work for AI as well.